drivers/dali/dali_lpc11u6x.c

```diff
@@ -0,0 +1,1187 @@
+	counter_tx_set(DALI_TX_IDLE);
+	counter_rx_start();
+	if (mcu_rx_pin()) {
+		data->rx_status = IDLE;
+	} else {
+		data->rx_status = BUS_LOW;
+		counter_rx_set_event_counter(STOPBIT, FAILURE_CONDITION_US);
+		counter_rx_event_enable(STOPBIT, true);
+	}
+	return 0;
+}
+
+static int dali_lpc11u6x_receive(const struct device *dev, struct dali_frame *frame,
+				 k_timeout_t timeout)
+{
+	__ASSERT(frame, "invalid frame pointer");
+	__ASSERT(dev, "invalid device");
+
+	struct dali_lpc11u6x_data *data = dev->data;
+	__ASSERT(&(data->rx_queue), "invalid queue");
+
+	if (k_msgq_get(&(data->rx_queue), frame, timeout) < 0) {
+		return -ENOMSG;
+	}
+	return 0;
+};
+
+static int dali_lpc11u6x_send(const struct device *dev, const struct dali_tx_frame *tx_frame)
+{
+	LOG_DBG("");
+	__ASSERT(dev, "invalid device");
+
+	const enum dali_event_type frame_type = tx_frame->frame.event_type;
+	if (frame_type == DALI_EVENT_NONE) {
+		return 0;
+	}
+
+	struct dali_lpc11u6x_data *data = dev->data;
+	if (data->active) {
+		if (data->active->index_next) {
+			LOG_ERR("send is busy sending");
+			return -EBUSY;
+		}
+	}
+
+	switch (frame_type) {
+	case DALI_FRAME_BACKWARD:
+	case DALI_FRAME_CORRUPT:
+		if (tx_frame->is_query) {
+			return -EINVAL;
+		}
+		if (is_tx_slot_empty(&data->backward)) {
+			data->rx_status = TRANSMIT_BACKFRAME;
+			counter_rx_event_enable(STOPBIT, false);
+			calculate_counts(dev, &data->backward, tx_frame->frame);
+			data->backward.inter_frame_idle = settling_time_us[0];
+		} else {
+			LOG_ERR("backward frame slot is busy");
+			return -EBUSY;
+		}
+		break;
+	case DALI_FRAME_DEVICE:
+	case DALI_FRAME_GEAR:
+	case DALI_FRAME_FIRMWARE:
+		if (is_tx_slot_empty(&data->forward)) {
+			calculate_counts(dev, &data->forward, tx_frame->frame);
+			if (tx_frame->priority < DALI_PRIORITY_1 ||
+			    tx_frame->priority > DALI_PRIORITY_5) {
+				return -EINVAL;
+			}
+			data->forward.inter_frame_idle = settling_time_us[tx_frame->priority];
+			data->forward.is_query = tx_frame->is_query;
+		} else {
+			LOG_ERR("forward frame slot is busy");
+			return -EBUSY;
+		}
+		break;
+	default:
+		return -EINVAL;
+	}
+	tx_schedule_send(data);
+
+	return 0;
+}
+
+static void dali_lpc11u6x_abort(const struct device *dev)
+{
+	__ASSERT(dev, "invalid device");
+
+	counter_tx_timer_stop();
+	counter_tx_set(DALI_TX_ACTIVE);
+
+	struct dali_lpc11u6x_data *data = dev->data;
+	tx_slot_reset_all(data);
+}
+
+static const struct dali_driver_api dali_lpc11u6x_driver_api = {
+	.receive = dali_lpc11u6x_receive,
+	.send = dali_lpc11u6x_send,
+	.abort = dali_lpc11u6x_abort,
+};
+
+/* forward declaration of init */
+static int init_dali_lpc11u6x(const struct device *dev);
+
+/* clang-format off */
+#define DALI_LPC11U6X_INST(id)                                                          \
+	static struct dali_lpc11u6x_data dali_lpc11u6x_data_##id = {0};                     \
+	static struct dali_lpc11u6x_config dali_lpc11u6x_config_##id = {                    \
```

Comment:
Config should be `const`

```suggestion
	static const struct dali_lpc11u6x_config dali_lpc11u6x_config_##id = {                    \
```